### ISSUE TITLE
Add Reps_And_Duration for type?

### ISSUE THREAD

**Issue by vfxdee** [IgnisDa/ryot#1157]
Could you add reps_and_duration and reps_and_duration_and_distance as types? If we're using this to track swimming laps we could use reps as a replacement for laps, then could track duration for the time per lap or total time for that set. We might also want to include distance swam in that as well. Is that a hard addition to make?

**Comment by IgnisDa** [comment]
Not very difficult, but I have a few questions.

<img width="863" alt="image" src="https://github.com/user-attachments/assets/21f99c5c-b88e-45e9-9c62-7faba300cc2e" />

What personal best types would be associated with these exercise types? Also what new best types can be calculated for them?

**Comment by vfxdee** [comment]
Hmm, I think reps/laps is definitely a thing. Time would be something. Could be time per lap or total time on laps. So best time would likely be shortest time per lap/rep (like total time divided by reps/laps = time per lap if not setting laps as sets or something). Volume/number of laps.

**Comment by IgnisDa** [comment]
Whats an exercise for reps_and_duration_and_distance?

**Comment by vfxdee** [comment]
Laps in a pool could be laps and duration or laps and duration and distance since you could also track distance swam. Running in a track could be the same. Admittedly, I'd use laps and duration more. I'm not much for a track or total distance travelled.

**Comment by IgnisDa** [comment]
Makes sense. But I won't have any of the inputs nullable. If you need that, you should create two different exercises.
